tests/integration/test_managers_integration.py

```
@@ -0,0 +1,439 @@
+class TestManagerPerformance:
+    """Test performance aspects of managers."""
+    
+    def test_manager_performance_large_registry(self):
+        """Test manager performance with large registry."""
+        manager = DatasetManager()
+        
+        # Register many components
+        for i in range(100):
+            class MockLoader:
+                def __init__(self):
+                    self.name = f"loader_{i}"
```

Ruff (0.13.1)

Function definition does not bind loop variable `i`

(B023)